_base/lang.js

```diff
@@ -307,7 +307,7 @@ define(["./kernel", "../has", "../sniff"], function(dojo, has){
 			// summary:
 			//		Returns true if it is a built-in function or some other kind of
 			//		oddball that *should* report as a function but doesn't
-			return it && !lang.isFunction(it) && /\{\s*\[native code\]\s*\}/.test(String(it)); // Boolean
+			return !!it && lang.isFunction(it) && /\{\s*\[native code\]\s*\}/.test(String(it)); // Boolean
```

Comment:
Judging from the text in https://bugs.dojotoolkit.org/ticket/18244, and also by @mlabrec's initial comment to this PR "**Also** made the function return bool for falsy arguments.", I think the main point was to change the `!isFunction()` to `isFunction()`.

And I assume that's because the bug/PR author didn't carefully read the spec for what `isAlien()` is supposed to do.  It's called `isAlien()`, not `isNativeOrNormalFunction()`.